### ISSUE TITLE
Patient count feature 

### ISSUE THREAD

**Issue by f-odhiambo** [opensrp/fhircore#141]
This functionality will allow you to know the total number of clients enrolled on the app once all patient data is synced. 
This information will be displayed as a number on the navigation drawer as displayed in the design attached

**Comment by f-odhiambo** [comment]
<img width="381" alt="Screen Shot 2021-05-25 at 10 39 24 AM" src="https://user-images.githubusercontent.com/4540684/119458530-86fc3200-bd45-11eb-8705-d8b8dfbaea6b.png">

**Comment by maimoonak** [comment]
There are few points need clarification
1- The item on drawer rightnow does not have any action associated with.. Clicking on the item should reload the activity?
2- The count not necessarily have to be updated only after sync... rather it would be count of patients into database which can be updated after sync or even after local registration... right!
3- The drawer rightnow is tightly coupled with Patient activity i.e. init into activity code. I am implementing a Base-Activity which would load drawer, nav items etc... Ok?

**Comment by maimoonak** [comment]
https://www.hl7.org/fhir/search.html#summary this allows to query summary for searches without loading whole bunch of records

**Comment by rkodev** [comment]
@maimoonak 
1. The item on drawer rightnow does not have any action associated with.. Clicking on the item should reload the activity?
**The item should navigate to another activity in case multiple items are in the drawer but clicking on the menu item when Activity / Fragment is displayed should redraw the activity**
2. The count not necessarily have to be updated only after sync... rather it would be count of patients into database which can be updated after sync or even after local registration... right!
**Yes**
3. The drawer rightnow is tightly coupled with Patient activity i.e. init into activity code. I am implementing a Base-Activity which would load drawer, nav items etc... Ok?
**This would be okay**

At @maimoonak This should be okay, returning a summary object with the given details

**Comment by maimoonak** [comment]
1- is there any reference doc on how to use FHIR engine?
2- By FHIR search standards we can pass some additional params into searches like _language, _summary, _include etc. https://www.hl7.org/fhir/search.html#summary. Based on type of param it can return a different result including summary, details or any other. 

I can not figure out how to pass a param to fhir engine such that  it returns count only ( like we do Patient.ADDRESS _ CITY can pass Resource._SUMMARY but seems that it does not allow it.

GET - http://hapi.fhir.org/baseR4/Patient?given=joh&_pretty=true&_summary=count
PAYLOAD -
```
 {
    "resourceType": "Bundle",
    "id": "7ff5baac-5e84-4249-b49a-be3b93fecb1e",
    "meta": {
        "lastUpdated": "2021-05-26T11:36:26.549+00:00",
        "tag": [
            {
                "system": "http://terminology.hl7.org/CodeSystem/v3-ObservationValue",
                "code": "SUBSETTED",
                "display": "Resource encoded in summary mode"
            }
        ]
    },
    "type": "searchset",
    "total": 2092
}
```

In short we want to find a good way of displaying summaries of all resources (count of patients, Encounters etc) via a count query rather than picking list size. 

PR is here which rightnow implements count via list size https://github.com/opensrp/fhircore/pull/161/files#diff-c5c92419f4716e1a738c367afb67986c2af093f8d9899a3ea88ca9b7665409c8R43
